com.ibm.wala.core/src/com/ibm/wala/ipa/callgraph/AnalysisScopeAppExclusions.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * ***************************************************************************** Copyright (c) 2007
+ * IBM Corporation. All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * <p>Contributors: IBM Corporation - initial API and implementation
+ * *****************************************************************************
+ */
+package com.ibm.wala.ipa.callgraph;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import com.ibm.wala.classLoader.IClass;
+import com.ibm.wala.classLoader.Language;
+import com.ibm.wala.types.ClassLoaderReference;
+import com.ibm.wala.util.config.SetOfClasses;
+
+/**
+ * Class to specify an analysis scope where some classes loaded by the
+ * application loader will not be considered "application" classes. This is
+ * useful when analyzing programs where library code has been added into the jar
+ * and thus would normally be treated as "application" code by WALA.
+ */
+public class AnalysisScopeAppExclusions extends AnalysisScope {
+
+  private SetOfClasses applicationExclusions;
+
+  public static AnalysisScopeAppExclusions createJavaAnalysisScope(SetOfClasses applicationExclusions) {
+    AnalysisScopeAppExclusions scope = new AnalysisScopeAppExclusions(Collections.singleton(Language.JAVA), applicationExclusions);
+    scope.initForJava();
+    return scope;
+  }
+
+  protected AnalysisScopeAppExclusions(Collection<? extends Language> languages, SetOfClasses applicationExclusions) {
```

Comment:
This is analogous to the AnalysisScope (superclass) constructor, see AnalysisScope.java line 132. I am not sure if we should change it in the subclass given that the superclass does it that way.